setup-dev.sh

```diff
@@ -5,6 +5,25 @@ echo "# Install testing tools."
 echo "# Only works with Ubuntu / APT."
 echo "# --------------------------------------"
 
+hook_script_name=".git/hooks/pre-push"
+
+if [ -t 1 ]; then
+    while true; do
```

Comment:
We don't need this `while true` loop, and we can remove the `break` statements below. `;;` is all you need to break out of a `case` statement in bash.